### PR TITLE
Rate limit otel errors

### LINKED ISSUE #2953: Opentelemetry error logs should be rate limited
Currently if an opentelemetry exporter gets overloaded the error handler will log errors very rapidly. This needs to be throttled somehow.

Suggest every 10 seconds based on the type of message.

### PR DESCRIPTION
**Pull request by BrynCooke**
When a batch span exporter is unable to send accept a span because the buffer is full it will emit an error.
These errors can be very frequent and could potentially impact performance.

Otel errors are not rate limited to one every ten seconds per error type.

Fixes #2953

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]. It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]. Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]. Tick whichever testing boxes are applicable. If you are adding Manual Tests:
    - please document the manual testing (extensively) in the Exceptions.
    - please raise a separate issue to automate the test and label it (or ask for it to be labeled) as `manual test`

### CODE REVIEW

**Review by bnjjj — APPROVED** [review]

File: .changesets/fix_payday_floor_search_calcium.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+### Rate limit errors emitted from open telemetry ([Issue #2953](https://github.com/apollographql/router/issues/2953))
+
+When a batch span exporter is unable to send accept a span because the buffer is full it will emit an error.
+These errors can be very frequent and could potentially impact performance.
+
+Otel errors are not rate limited to one every ten seconds per error type.
```
Comment (BrynCooke) [inline .changesets/fix_payday_floor_search_calcium.md:6]:
```suggestion
Otel errors are now rate limited to one every ten seconds per error type.
```

**Comment by BrynCooke** [comment]
Perf test please

**Comment by BrynCooke** [comment]
Perf test please

**Comment by BrynCooke** [comment]
Perf test please

**Comment by BrynCooke** [comment]
Perf test please

**Comment by BrynCooke** [comment]
Perf test please

**Comment by BrynCooke** [comment]
Perf test please

**Comment by BrynCooke** [comment]
Perf test please

**Comment by BrynCooke** [comment]
Perf test please

**Pull request merged**